core/src/dht/actions/hold.rs

```diff
@@ -4,28 +4,28 @@ use crate::{
     action::{Action, ActionWrapper},
     context::Context,
     instance::dispatch_action,
+    network::entry_with_header::EntryWithHeader,
 };
 use futures::{
     future::Future,
     task::{LocalWaker, Poll},
 };
 use holochain_core_types::{
     cas::content::{Address, AddressableContent},
-    entry::Entry,
     error::HolochainError,
 };
 use std::{pin::Pin, sync::Arc};
 
 pub async fn hold_entry<'a>(
-    entry: &'a Entry,
-    context: &'a Arc<Context>,
+    entry_wh: EntryWithHeader,
+    context: Arc<Context>,
 ) -> Result<Address, HolochainError> {
-    let action_wrapper = ActionWrapper::new(Action::Hold(entry.clone()));
+    let action_wrapper = ActionWrapper::new(Action::Hold(entry_wh.clone()));
     dispatch_action(context.action_channel(), action_wrapper.clone());
 
     await!(HoldEntryFuture {
-        context: context.clone(),
-        address: entry.address(),
+        context: context,
+        address: entry_wh.entry.address(),
```

Comment:
don't collapse names of variables like this, write the whole thing out :P

Reply:
How about we deal with this after the `ChainItem` refactor?